library/src/main/java/dev/chrisbanes/insetter/Insetter.java

```diff
@@ -67,8 +210,12 @@ public static void setOnApplyInsetsListener(
           @Override
           public WindowInsetsCompat onApplyWindowInsets(View v, WindowInsetsCompat insets) {
             listener.onApplyInsets(v, insets, initialState);
-            // Always return the initial insets instance
-            return insets;
+
+            if (consumeSystemWindowInsets) {
+              return insets.consumeSystemWindowInsets();
+            } else {
+              return insets;
+            }
```

Comment:
I also considered surfacing this so callers can be more granular with how the consume insets.

Reply:
Yep, lgtm

Reply:
In that case, would it make sense to have `OnApplyInsetsListener.onApplyInsets` return `WindowInsetsCompat`? The resulting extension could then look like this:

```
inline fun View.doOnApplyWindowInsets(
    crossinline f: (view: View, insets: WindowInsetsCompat, initialState: ViewState) -> WindowInsetsCompat
) = Insetter.builder()
    .setOnApplyInsetsListener { view, insets, initialState ->
        f(view, insets, initialState)
    }.applyToView(this)
```

Reply:
This also seems like it would make testing `Builder.consumeSystemWindowInsets` a bit easier